### PR TITLE
Feature/buffered message sending

### PR DESCRIPTION
**Pull request by hebein187**

### CODE REVIEW

File: src/main/java/com/dynatrace/prototype/SlackMessageSenderService.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public void run() {
+        Slack slack = Slack.getInstance();
+        Iterator<OffsetDateTime> keyIterator = new TreeSet<>(bufferedPostMessages.keySet()).iterator(); //due to the TreeSet the times are sorted by the natural order which is ascending so the earliest time is first
+
+        while (keyIterator.hasNext()) {
+            try {
+                OffsetDateTime currentKey = keyIterator.next();
+                ChatPostMessageRequest request = bufferedPostMessages.get(currentKey);
+
+                bufferedPostMessages.remove(currentKey);
```
Comment (thisthat) [inline src/main/java/com/dynatrace/prototype/SlackMessageSenderService.java]: I would move these lines before the `try`

File: src/main/java/com/dynatrace/prototype/payloadCreator/SlackCreator.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 public class SlackCreator {
     public static final String SLACK_STYLE_PRIMARY = "primary";
     public static final String SLACK_STYLE_DANGER = "danger";
-    private static final String COLOR_PASS = "#00FF00";
+    private static final String COLOR_PASS = "#00FF00"; //TODO: ask Giovanni if should be configurable and and all methods not static because to configure a constructor is needed
```
Comment (thisthat) [inline src/main/java/com/dynatrace/prototype/payloadCreator/SlackCreator.java]: I think you can remove this TODO :)

File: src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java [hunk 3]
```diff
@@ -59,18 +63,38 @@ public SlackHandler() {
...
         mappers.add(new ProblemMapper());
+
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        String token = System.getenv(ENV_SLACK_TOKEN);
+
+        if (token == null) {
+            System.err.println(ENV_SLACK_TOKEN + " is null!");
+        } else {
+            String intervalString = System.getenv(ENV_SLACK_MSG_INTERVAL);
+            int interval = 10000; //TODO: better default value in program or env variable with default value in deployment?
```
Comment (thisthat) [inline src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java]: the former 😉

File: src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java [hunk 4]
```diff
@@ -83,14 +107,17 @@ public boolean receiveEvent(KeptnCloudEvent event) {
...
 
                 ChatPostMessageRequest request = SlackCreator.createPostRequest(event, channel, layoutBlocks, SLACK_NOTIFICATION_MSG);
-                ChatPostMessageResponse response = slack.methods(token).chatPostMessage(request);
+                OffsetDateTime requestKey = OffsetDateTime.parse(event.getTime());
 
-                if (response.isOk()) {
+                bufferedPostMessages.put(requestKey, request);
+
+                if(request.equals(bufferedPostMessages.get(requestKey))) {
+                    System.out.println("Post message added!");
```
Comment (thisthat) [inline src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java:114]: We should add a logging framework as a follow up task :)

**Review by thisthat — APPROVED** [review]

File: src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java (hunk 4 above)
Comment (hebein187) [reply src/main/java/com/dynatrace/prototype/payloadHandler/SlackHandler.java:114]: This is a future issue for probably the next internship.

**Pull request merged**